sklearn/tree/export.py

```diff
@@ -778,3 +780,171 @@ def export_graphviz(decision_tree, out_file=None, max_depth=None,
     finally:
         if own_file:
             out_file.close()
+
+
+def export_ascii(decision_tree, feature_names=None,
+                 max_depth=10, show_value='only_leaves',
+                 show_class=True, spacing=3):
+    """Build a text report showing the rules of a decision tree.
+
+    Parameters
+    ----------
+    decision_tree : object
+        The decision tree estimator to be exported.
+        It can be an instance of
+        DecisionTreeClassifier or DecisionTreeRegressor.
+
+    feature_names : list, optional (default=None)
+        A list of length n_features containing the feature names.
+        If None generic names will be used ("feature_0", "feature_1", ...).
+
+    max_depth : int, optional (default=10)
+        Only the first max_depth levels of the tree are exported.
+
+    show_value : string or None, optional (default='only_leaves')
+        Wheter to report the predicted value (for regression) or
+        the class weights (for classification).
+        If 'only_leaves' the value will be exported only for the leaves;
+        If 'all' the value will be exported for all the nodes;
+        If None, no value will be exported.
+
+    show_class : bool, optional (default=True)
+        If False the class label is not exported for each node.
+        The class is exported by default for classification trees.
+        Only relevant for classification problems.
+
+    spacing : int, optional (default=3)
+        Number of spaces between edges. The higher it is, the wider the result.
+
+    Returns
+    -------
+    report : string
+        Text summary of all the rules in the decision tree.
+
+    Examples
+    -------
+
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.tree import DecisionTreeClassifier
+    >>> from sklearn.tree.export import export_ascii
+    >>> iris = load_iris()
+    >>> X = iris['data']
+    >>> y = iris['target']
+    >>> decision_tree = DecisionTreeClassifier(random_state=0, max_depth=2)
+    >>> decision_tree = decision_tree.fit(X, y)
+    >>> r = export_ascii(decision_tree, feature_names=iris['feature_names'])
+    >>> print(r)
+    |---petal width (cm) <= 0.80
+    |   | (class: 0)
+    |   |---* (weights: [50.0, 0.0, 0.0])
```

Comment:
I don't really get why weights is in (). And I think `class: 0` should appear on the same line, e.g. `weights: [50.0, 0.0, 0.0] => class: 0`. Or else:
```
| weights: [50.0, 0.0, 0.0]
|---* class: 0
```
?